workspaces/adventure-pack/goodies/typescript/Map.groupBy/index.test.ts

```diff
@@ -0,0 +1,157 @@
+import { describe, expect, it } from "@jest/globals";
+import "./index";
+
+describe("Map.groupBy", () => {
+  it("should group elements by the result of the callback function", () => {
+    const numbers = [1, 2, 3, 4, 5, 6];
+    const result = Map.groupBy(numbers, (num) =>
+      num % 2 === 0 ? "even" : "odd",
+    );
+
+    expect(result.size).toBe(2);
+    expect(result.get("even")).toStrictEqual([2, 4, 6]);
+    expect(result.get("odd")).toStrictEqual([1, 3, 5]);
```

Comment:
If you felt like it, I believe `toStrictEqual` accepts other built-in types, though I don't have a strong opinion about whether this is nicer to work with in any way:

```ts
expect(result).toStrictEqual(new Map([
  ["even", [2, 4, 6]],
  ["odd", [1, 3, 5]],
]));

Reply:
Yeah I think it's nicer, updated for consistency most places.